internal/generator/vector/input/viaq_test.go

```diff
@@ -79,6 +79,33 @@ var _ = Describe("inputs", func() {
 		},
 			"viaq_application_with_includes_excludes.toml",
 		),
+		Entry("with an application that specs infra namespaces and exclude namespaces", logging.InputSpec{
+			Name: "my-app",
+			Application: &logging.Application{
+				Namespaces:        []string{"test-ns-foo", "openshift-logging", "kube-apiserver"},
+				ExcludeNamespaces: []string{"test-ns1"},
+			},
+		},
+			"viaq_application_with_infra_includes_excludes.toml",
+		),
+		Entry("with an application that specs infra namespaces and excludes infra namespaces", logging.InputSpec{
+			Name: "my-app",
+			Application: &logging.Application{
+				Namespaces:        []string{"test-ns-foo", "openshift*", "kube-apiserver"},
+				ExcludeNamespaces: []string{"test-ns1", "openshift-logging"},
+			},
+		},
+			"viaq_application_with_infra_includes_infra_excludes.toml",
+		),
+		Entry("with an application that specs specific infra namespace and excludes infra namespaces", logging.InputSpec{
```

Comment:
This is an interesting test case because I think we will get this one alot:  "why do I not get logs from my openshift-logging" namespace..... its because the excludes take precedence to the includes

Reply:
Basically from the [Vector docs](https://vector.dev/docs/reference/configuration/sources/file/#exclude)

> [excludes] Takes precedence over the include option. Note: The exclude patterns are applied after the attempt to glob everything in include. This means that all files are first matched by include and then filtered by the exclude patterns. This can be impactful if include contains directories with contents that are not accessible.